test/uri_internal_test.c

```diff
@@ -0,0 +1,522 @@
+        fprintf(stderr, "  query     = '%s' (expected '%s')\n", query,
+                test_data->expected.query);
+        fprintf(stderr, "  fragment  = '%s' (expected '%s')\n", fragment,
+                test_data->expected.fragment);
+        fprintf(stderr, "  user      = '%s' (expected '%s')\n", user,
+                test_data->expected.user);
+        fprintf(stderr, "  password  = '%s' (expected '%s')\n", password,
+                test_data->expected.password);
+        fprintf(stderr, "  host      = '%s' (expected '%s')\n", host,
+                test_data->expected.host);
+        fprintf(stderr, "  service   = '%s' (expected '%s')\n", service,
+                test_data->expected.service);
+    } else {
+        fprintf(stderr, "Test %d OK\n", testnum);
+        fprintf(stderr, "  %s    = '%s'\n",
+                test_data->good ? "uri   " : "baduri", test_data->uri);
+        if (scheme || test_data->expected.scheme)
+            fprintf(stderr, "  scheme    = '%s' (expected '%s')\n", scheme,
+                    test_data->expected.scheme);
+        if (authority || test_data->expected.authority)
+            fprintf(stderr, "  authority = '%s' (expected '%s')\n",
+                    authority, test_data->expected.authority);
+        if (path || test_data->expected.path)
+            fprintf(stderr, "  path      = '%s' (expected '%s')\n", path,
+                    test_data->expected.path);
+        if (query || test_data->expected.query)
+            fprintf(stderr, "  query     = '%s' (expected '%s')\n", query,
+                    test_data->expected.query);
+        if (fragment || test_data->expected.fragment)
+            fprintf(stderr, "  fragment  = '%s' (expected '%s')\n",
+                    fragment, test_data->expected.fragment);
+        if (user || test_data->expected.user)
+            fprintf(stderr, "  user      = '%s' (expected '%s')\n",
+                    user, test_data->expected.user);
+        if (password || test_data->expected.password)
+            fprintf(stderr, "  password  = '%s' (expected '%s')\n",
+                    password, test_data->expected.password);
+        if (host || test_data->expected.host)
+            fprintf(stderr, "  host      = '%s' (expected '%s')\n",
+                    host, test_data->expected.host);
+        if (service || test_data->expected.service)
+            fprintf(stderr, "  service   = '%s' (expected '%s')\n",
+                    service, test_data->expected.service);
+    }
+ err:
+    ERR_clear_error();
+    OPENSSL_free(scheme);
+    OPENSSL_free(authority);
+    OPENSSL_free(path);
+    OPENSSL_free(query);
+    OPENSSL_free(fragment);
+    OPENSSL_free(user);
+    OPENSSL_free(password);
+    OPENSSL_free(host);
+    OPENSSL_free(service);
+
+    return errcount;
+}
+
+int main(int argc, char *argv[])
+{
+    FILE *f;
+    TEST_DATA *test_data = NULL;
+    int errs = 0;
+
+    OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CRYPTO_STRINGS, NULL);
```

Comment:
'cause they aren't loaded by default.  But maybe that should be done in `crypto/uri.c`, instead of `OPENSSL_init_crypto(0, NULL);`?